buildSrc/src/main/groovy/BuildkitBuild.groovy

```diff
@@ -0,0 +1,399 @@
+            logger.quiet("Using Release Type : '${releaseType}")
+        }
+    }
+
+    @Override
+    @TaskAction
+    protected void exec() {
+        def buildBaseDir = temporaryDir.toPath()
+        def containerizationDir = Paths.get("$buildBaseDir/containerization/")
+        def driverDir = Paths.get("$buildBaseDir/jdbc-driver/")
+        def containerLocation = '/opt/override/'
+        def driverLocation = '/opt/jdbc-driver'
+        def sourceFiles = sourceTasks.get().collect { it -> it.getOutputs().files.files }.flatten() as List<File>
+        String gitRevision = gitTask.flatMap { it.revision }.get().replace("\n", "")
+        def imageRepo = generateDockerTagging(gitRevision)
+        List<String> dockerAuth = new ArrayList(['docker-remotes.software.r3.com', 'corda-os-docker.software.r3.com'])
+
+        workingDir project.rootDir
+
+        if (!(Files.exists(containerizationDir))) {
+            logger.quiet("Created containerization dir")
+            Files.createDirectories(containerizationDir)
+        }
+
+        if (!(Files.exists(driverDir))) {
+            logger.quiet("Created jdbc-driver dir")
+            Files.createDirectories(driverDir)
+        }
+
+        (sourceFiles + extraSourceFiles).forEach {
+            if (Files.exists(Paths.get(it.path))) {
+                Files.copy(Paths.get(it.path), Paths.get("${containerizationDir.toString()}/$it.name"), StandardCopyOption.REPLACE_EXISTING)
+            }
+        }
+
+        jdbcDriverFiles.forEach {
+            if (Files.exists(Paths.get(it.path))) {
+                Files.copy(Paths.get(it.path), Paths.get("${driverDir.toString()}/$it.name"), StandardCopyOption.REPLACE_EXISTING)
+            }
+        }
+
+        executeBuildkitCommand(containerLocation, dockerAuth, imageRepo, containerizationDir, driverDir, driverLocation)
+    }
+
+    /**
+     * helper method to build and execute the buildkit command
+     * @param containerLocation
+     * @param dockerAuth list of repositories to log into
+     * @param imageRepo list of repositories to publish to
+     * @param containerizationDir
+     * @param driverDir
+     * @param driverLocation
+     */
+    def executeBuildkitCommand(containerLocation, List<String> dockerAuth, ArrayList imageRepo, containerizationDir, driverDir, driverLocation) {
+        List<String> javaArgs = new ArrayList<String>(arguments.get())
+        javaArgs.add("-Dlog4j2.debug=\${ENABLE_LOG4J2_DEBUG:-false}")
+        javaArgs.add("-Dlog4j.configurationFile=log4j2-console.xml")
+        javaArgs.add("-Dpf4j.pluginsDir=${containerLocation + subDir.get()}")
+
+        def baseImageName = "${baseImageTag.get().empty ? baseImageName.get() : "${baseImageName.get()}:${baseImageTag.get()}"}"
+        def entryName = overrideEntryName.get().empty ? projectName : overrideEntryName.get()
+        
+        for(repo in imageRepo){
+            dockerAuth.add(repo.name)
+        }
+
+        dockerLogin(dockerAuth)
```

Comment:
Remember corda is open source, what happens if a non R3 local users hits this part of code ( you can temp change your Artifactory env variables and find out - but it will fail as they of course cant login to Artifactory).

This needs to be conditional.

see something like the check [here ](https://github.com/corda/corda-runtime-os/blob/release/os/5.0/buildSrc/src/main/groovy/DeployableContainerBuilder.groovy#L330)